types/matcher.d.ts

```diff
@@ -20,9 +20,14 @@ export type Match<T extends DefaultParams = DefaultParams> =
 
 export type MatcherFn = (pattern: Path, path: Path) => Match;
 
+export type MakeRegexpFn = {
```

Comment:
The type looks good. I would just rename it to 
```suggestion
export type PatternToRegexpResult = {
```